compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -98,19 +98,19 @@ fn eval_body_using_ecx<'tcx, R: InterpretationResult<'tcx>>(
             return Err(ecx
                 .tcx
                 .dcx()
-                .emit_err(errors::DanglingPtrInFinal { span: ecx.tcx.span, kind: intern_kind })
-                .into());
+                .emit_err(errors::DanglingPtrInFinal { span: ecx.tcx.span, kind: intern_kind }))
+            .into();
         }
         Err(InternResult::FoundBadMutablePointer) => {
             return Err(ecx
                 .tcx
                 .dcx()
-                .emit_err(errors::MutablePtrInFinal { span: ecx.tcx.span, kind: intern_kind })
-                .into());
+                .emit_err(errors::MutablePtrInFinal { span: ecx.tcx.span, kind: intern_kind }))
+            .into();
```

Comment:
These could be simplified by having direct From impls for the error types

Reply:
I don't follow? We can't return `Err(...)` since that has type `Result`, not `InterpResult`. So there has to be at least one outside `into()` after the `Err`, and that's the only thing we have here.

Reply:
I mean we could skip the `Err` and have a `From<ErrorGuaranteed>` and others directly on InterpResult

Reply:
Oh, that's what you mean... seems a bit unusual to have a `From` to directly turn an error into a *result* (rather than a different error)?